bento/src/main/java/com/yelp/android/bento/components/CarouselComponent.kt

```diff
@@ -69,17 +72,40 @@ open class CarouselComponentViewHolder : ComponentViewHolder<Unit?, CarouselView
 
     lateinit var recyclerView: RecyclerView
     lateinit var controller: RecyclerViewComponentController
+    lateinit var element: CarouselViewModel
     private var attachedPool = false
 
+    /**
+     * Saves the scroll position of the carousel to the view model so that it can be restored when
+     * the carousel recyclerView itself is recycled in the larger component list. Scroll position
+     * is saved when the scroll state of the carousel transitions to
+     * [RecyclerView.SCROLL_STATE_IDLE].
+     */
+    private fun saveScrollPosition() {
+        (recyclerView.layoutManager as? LinearLayoutManager)?.apply {
+            element.scrollPosition = findFirstVisibleItemPosition()
+            element.scrollPositionOffset =
+                    recyclerView.getChildAt(0).left - recyclerView.paddingLeft
```

Comment:
Is it possible for the recyclerview to be empty when this is called? (ie. cause an index out of bounds exception)

Reply:
I want to say it's not possible, since the only time we save the position is when the carousel scroll listener is triggered, and I think that's impossible to trigger on an empty recyclerview. I don't have any evidence of that though so I'll just add a test and a check to be safe.